jwst/outlier_detection/utils.py

```diff
@@ -66,13 +66,19 @@ def create_median(resampled_models, maskpt, on_disk=True, buffer_size=10.0):
     # compute median over all models
     if not on_disk:
         # easier case: all models in library can be loaded into memory at once
-        model_list = []
         with resampled_models:
-            for resampled in resampled_models:
-                model_list.append(resampled.data)
-                resampled_models.shelve(resampled, modify=False)
+            for i, resampled in enumerate(resampled_models):
+                # convert to a CubeModel to pass into create_cube_median
+                if i == 0:
+                    cube_shape = (len(resampled_models), resampled.data.shape[0], resampled.data.shape[1])
+                    cube = datamodels.CubeModel(data=np.zeros(cube_shape))
+                    cube.wht = np.zeros(cube_shape)
+
+                cube.data[i] = resampled.data
+                cube.wht[i] = resampled.wht
+                resampled_models.shelve(resampled, i, modify=False)
```

Comment:
```suggestion
            for i, resampled in enumerate(resampled_models):
                data = resampled.data.copy()
                data[resampled.wht < weight_threshold[i]] = np.nan
                model_list.append(data)
                resampled_models.shelve(resampled, i, modify=False)
```
I think something like this would handle the per-model weight (or be taken one step further and combine the above iteration through the models since the weight thresholds are per-model). It would require re-adding model_list and the nanmedian (which could used a warning filter to avoid a warning for all-nan slices).

Reply:
Thanks, I applied your suggestion and refactored a bit to avoid looping twice.  Let me know how it looks to you.